android/expoview/src/main/java/host/exp/exponent/test/TestActionEvent.kt

```diff
@@ -0,0 +1,12 @@
+// Copyright 2015-present 650 Industries. All rights reserved.
+package host.exp.exponent.test
+
+class TestActionEvent(
+  val id: Int,
+  val selectorType: String?,
+  val selectorValue: String?,
+  val actionType: String?,
+  val actionValue: String?,
```

Comment:
unfortunately no, they're still constructed with nullable values

Reply:
OK. I see what is happening. ExponentTestNativeModule, which constructs TestActionEvents, is written confusingly.

It receives an object from JS and checks the existence of some keys but not others. However, readableMap.getString() already performs a [hasKey](https://github.com/facebook/react-native/blob/1465c8f3874cdee8c325ab4a4916fda0b3e43bdb/ReactAndroid/src/main/java/com/facebook/react/bridge/ReadableNativeMap.java#L160-L163) check.

I imagine ExponentTestNativeModule's intent was to expect selectorType and actionType to be defined and for selectorValue and actionValue to be nullable. This said, we probably would want to change ExponentTestNativeModule to reject its promise if selectorType or actionType are null instead of using Kotlin's !! to throw.

Reply:
Will follow-up in a separate PR: https://linear.app/expo/issue/ENG-1835/follow-up-on-exponenttestnativemodule-selectortype-and-actiontype